### ISSUE TITLE
Cannot get live stream of objects using repeat() and autoMap()

### ISSUE THREAD

**Issue by adam-mahmood** [davidmoten/rxjava2-jdbc#3]
Hi david 
The automap feature works, thank you for the fix; sorry to be a pain but i am unable to retrieve live objects produced in the database via this code snippet:
```
db.select( "SELECT RecIndex,Site FROM dbo.MEAS_CARRIER where Time >= ? ORDER BY RecIndex ASC") .parameterStream(Flowable.just(Instant.now()).delay(1, TimeUnit.SECONDS).repeat())
.autoMap(CarrierMeasuremnt.class).distinct();
```

As you can see i am passing Instant.now() into the flowable and i am repeating it, so i should expect a live stream objects being produced as is the case if i use the get getAs(InputStream.class) function instead of autoMap().
 I believe the issue is to with the equals and hashcode methods of the ProxyInstance? Can you please clarify this?

**Comment by davidmoten** [comment]
Yes indeed you need to be careful with `distinct` for starters. Use the overload of `distinct` with the comparator instead so you can explicitly specify what distinct means.

The other thing to note is that repeating `Flowable.just(Instant.now())` will give you exactly the same flowable each time because Instant.now() is calcuated once only. What you need is `Flowable.fromCallable(()-> Instant.now())`.

**Comment by davidmoten** [comment]
By the way if you use RC4 distinct might work as is because I enhanced the equals and hashCode methods.

**Comment by adam-mahmood** [comment]
I am using `Flowable.fromCallable(()-> Instant.now())` now.
I have checked out your code and i am using RC5-SNAPSHOT. From the logging in class Select.Java on line 54 , it seems i am getting live data from the database however it does not proceed to emitting the value at line 58, so i believe the issue lies inside @ line 57  `T v = mapper.apply(rs);` where you are doing some magic to create the java objects which i can hardly follow.If you do think of something please tell, and yet again thank you for your swift reply

**Comment by adam-mahmood** [comment]
Hi David i am not using `distinct `and i still get the same issue. It seems the jdbc `ResultSet` is closing inside or just before the `values()` function in class `Util.java`. Do you have any ideas on how to resolve this?

**Comment by davidmoten** [comment]
Thanks again, I've got your behaviour in a unit test now and I'll sort it
out.

**Comment by davidmoten** [comment]
Beaut, you picked up a bug with reusing PreparedStatements in autoMap. I've fixed it now and deployed 0.1-RC5 to Maven Central. By the way I might do some more work soon on a nicer equals/hashCode so that distinct works on the actual mapped values. Perhaps tomorrow.

**Comment by adam-mahmood** [comment]
Am i right to say that the proxyservice was going out of scope thats why the resultset closed?

**Comment by adam-mahmood** [comment]
Its working now. Thank you very much David!

**Comment by adam-mahmood** [comment]
Another issue i need to point out is your maven definition in the `readme` is wrong. You have:
```
<dependency>
  <groupId>com.github.davidmoten.rxjava2-jdbc</groupId>
  <artifactId>rxjava2-jdbc</artifactId>
  <version>0.1-RC4</version>
</dependency>
```
i believe you added the artifactId at the end of the groupId, and if you look at the pom its: `<groupId>com.github.davidmoten.rxjava2-jdbc</groupId>`

Sorry to be a pain, i thought it would be good to let you know.

**Comment by davidmoten** [comment]
Ah glad to hear it's working now @adam-mahmood, and thanks for the correction for the README, that's not being a pain, absolutely the opposite!

**Comment by adam-mahmood** [comment]
Hi @davidmoten  does your TransactedSelectBuilder have an automap feature? I want to return rows from two tables (these rows are linked by id), any ideas on how this can be done with the existing library?

**Comment by davidmoten** [comment]
Not sure, maybe I overlooked that. I'll check in a couple of hours.

**Comment by davidmoten** [comment]
@adam-mahmood I've added support for `transacted()` automapping and getAs tuples. Released as 0.1-RC6.

**Comment by adam-mahmood** [comment]
Thanks David. quite impressed by swift response times. Much Appreciated.
